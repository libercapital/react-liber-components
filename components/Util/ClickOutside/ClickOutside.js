"use strict";

require("core-js/modules/web.dom.iterable");

require("core-js/modules/es6.array.iterator");

require("core-js/modules/es6.object.to-string");

require("core-js/modules/es6.string.iterator");

require("core-js/modules/es6.weak-map");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

require("core-js/modules/es7.symbol.async-iterator");

require("core-js/modules/es6.symbol");

require("core-js/modules/es6.object.define-property");

require("core-js/modules/es6.object.create");

require("core-js/modules/es6.object.set-prototype-of");

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; if (obj != null) { var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

var ClickOutside =
/*#__PURE__*/
function (_Component) {
  _inherits(ClickOutside, _Component);

  function ClickOutside() {
    var _getPrototypeOf2;

    var _temp, _this;

    _classCallCheck(this, ClickOutside);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _possibleConstructorReturn(_this, (_temp = _this = _possibleConstructorReturn(this, (_getPrototypeOf2 = _getPrototypeOf(ClickOutside)).call.apply(_getPrototypeOf2, [this].concat(args))), _this.setWrapperRef = function (node) {
      _this.wrapperRef = node;
    }, _this.handleClickOutside = function (event) {
      var onOutsideClick = _this.props.onOutsideClick;

      if (_this.wrapperRef && !_this.wrapperRef.contains(event.target) && onOutsideClick) {
        onOutsideClick();
      }
    }, _temp));
  }

  _createClass(ClickOutside, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      document.addEventListener('mousedown', this.handleClickOutside);
    }
  }, {
    key: "componentWillUnmount",
    value: function componentWillUnmount() {
      document.removeEventListener('mousedown', this.handleClickOutside);
    }
  }, {
    key: "render",
    value: function render() {
      return _react["default"].createElement("div", {
        ref: this.setWrapperRef
      }, this.props.children);
    }
  }]);

  return ClickOutside;
}(_react.Component);

ClickOutside.propTypes = {
  children: _propTypes["default"].oneOfType([_propTypes["default"].any]).isRequired,
  onOutsideClick: _propTypes["default"].func
};
ClickOutside.defaultProps = {
  onOutsideClick: function onOutsideClick() {}
};
var _default = ClickOutside;
exports["default"] = _default;