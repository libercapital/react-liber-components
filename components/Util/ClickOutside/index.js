"use strict";

require("core-js/modules/es6.object.define-property");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "default", {
  enumerable: true,
  get: function get() {
    return _ClickOutside["default"];
  }
});

var _ClickOutside = _interopRequireDefault(require("./ClickOutside"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }