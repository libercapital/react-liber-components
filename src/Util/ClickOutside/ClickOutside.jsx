import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';


class ClickOutside extends Component {
  componentDidMount() {
    document.addEventListener('mousedown', this.handleClickOutside);
  }

  componentWillUnmount() {
    document.removeEventListener('mousedown', this.handleClickOutside);
  }

  setWrapperRef = (node) => {
    this.wrapperRef = node;
  }

  handleClickOutside = (event) => {
    const { onOutsideClick } = this.props;
    if (this.wrapperRef && !this.wrapperRef.contains(event.target) && onOutsideClick) {
      onOutsideClick();
    }
  }

  render() {
    return (
      <div ref={this.setWrapperRef}>
        {this.props.children}
      </div>
    );
  }
}
ClickOutside.propTypes = {
  children: PropTypes.oneOfType([PropTypes.any]).isRequired,
  onOutsideClick: PropTypes.func,
};
ClickOutside.defaultProps = {
  onOutsideClick: () => {},
};

export default ClickOutside;
