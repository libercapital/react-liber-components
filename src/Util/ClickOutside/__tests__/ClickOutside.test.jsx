import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import ClickOutside from '../ClickOutside';

describe('Testing ClickOutside Component', () => {
  let props;
  configure({ adapter: new Adapter() });
  beforeEach(() => {
    props = {
      onOutsideClick: sinon.spy(),
    };
  });
  it('should render correctly', () => {
    const wrapper = shallow(<ClickOutside {...props}><span>Children</span></ClickOutside>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should call function when click outside', () => {
    const map = {};
    document.addEventListener = jest.fn().mockImplementation((event, cb) => {
      map[event] = cb;
    });
    const wrapper = mount(<ClickOutside {...props}><span>Children</span></ClickOutside>);
    expect(wrapper.find('span').length).toBe(1);
    map.mousedown({ event: { target: 'mock' } });
    expect(props.onOutsideClick.called).toBeTruthy();
    wrapper.unmount();
  });
});
