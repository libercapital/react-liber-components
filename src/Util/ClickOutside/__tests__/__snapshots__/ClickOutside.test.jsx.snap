// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ClickOutside Component should render correctly 1`] = `
<div>
  <span>
    Children
  </span>
</div>
`;
